rocks-lib/src/tree/mod.rs

```diff
@@ -0,0 +1,58 @@
+use std::path::PathBuf;
+
+use crate::config::LuaVersion;
+use eyre::Result;
+
+/// A tree is a collection of files where installed rocks are located.
+///
+/// `rocks` diverges from the traditional hierarchy employed by luarocks.
+/// Instead, we opt for a much simpler approach:
+///
+/// - /rocks/<lua-version> - contains rocks
+/// - /rocks/<lua-version>/<rock>/etc - documentation and supplementary files for the rock
+/// - /rocks/<lua-version>/<rock>/lib - shared libraries (.so files)
+/// - /rocks/<lua-version>/<rock>/src - library code for the rock
+/// - /bin - binary files produced by various rocks
+
+pub struct Tree<'a> {
+    /// The Lua version of the tree.
+    version: &'a LuaVersion,
+    /// The root of the tree.
+    root: &'a PathBuf,
+}
+
+/// Change-agnostic way of referencing various paths for a rock.
+pub struct TreeLayout {
+    pub etc: PathBuf,
+    pub lib: PathBuf,
+    pub src: PathBuf,
```

Comment:
suggestion: Perhaps these should be `Option`s, since not every rock has all of these paths?

Reply:
I'm slightly inclined to create all of these anyway just for consistency's sake. Then external apps don't need to check for the existence of the directory before they try to access any content inside (makes globbing across all rocks simpler too). It's not like a single directory takes up gigabytes of storage.

I'm impartial to this change, so we can go one way or the other here :p

Reply:
how would you create them then?
just empty? like this external apps just would have to check if the contents exists instead of checking if the folder exists